core/unit_test/TestViewCopy_c.hpp

```diff
@@ -0,0 +1,359 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.2
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef TESTVIEWCOPY_C_HPP
+#define TESTVIEWCOPY_C_HPP
+
+#include <gtest/gtest.h>
+
+#include <Kokkos_Core.hpp>
+
+#include <array>
+#include <utility>
+
+namespace Test {
+
+namespace {
+// Do not rely on deep_copy(0) as we want to test it!
+template <class ViewType, class ExecSpace>
+void reset_view(const ExecSpace& space, ViewType& a, int magic) {
+  auto policy = Kokkos::RangePolicy<ExecSpace>(space, 0, a.span());
+
+  Kokkos::fence();
+  Kokkos::parallel_for(
+      "TestViewCopy::ResetView", policy,
+      KOKKOS_LAMBDA(const int i) { a.data()[i] = magic; });
+  space.fence();
+}
+
+template <typename ExecSpace, typename DT, typename... DP>
+bool check_magic_value(
+    const ExecSpace& space, const Kokkos::View<DT, DP...>& a, int magic,
+    std::enable_if_t<Kokkos::ViewTraits<DT, DP...>::rank == 1>* = nullptr) {
+  auto policy = Kokkos::RangePolicy<ExecSpace>(space, 0, a.extent(0));
+
+  bool all_elements_are_set = true;
```

Comment:
The initial value is overwritten, I would just not initialize it to avoid confusion.
```suggestion
  bool all_elements_are_set;
```